File: Flatshare Code/FlatShare/FlatshareCentral/src/main/java/com/joinflatshare/ui/profile/create/ProfileCreateActivity.java
package com.joinflatshare.ui.profile.create;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.Nullable;

import com.joinflatshare.FlatShareApplication;
import com.joinflatshare.FlatshareCentral.databinding.ActivityProfileCreateBinding;
import com.joinflatshare.api.retrofit.ApiManager;
import com.joinflatshare.chat.ApplicationChatHandler;
import com.joinflatshare.chat.SendBirdUser;
import com.joinflatshare.constants.AppConstants;
import com.joinflatshare.customviews.interests.InterestsView;
import com.joinflatshare.interfaces.OnUiEventClick;
import com.joinflatshare.pojo.user.Loc;
import com.joinflatshare.pojo.user.ModelLocation;
import com.joinflatshare.pojo.user.User;
import com.joinflatshare.pojo.user.UserResponse;
import com.joinflatshare.ui.base.gpsfetcher.GpsHandlerCallback;
import com.joinflatshare.ui.register.RegisterBaseActivity;
import com.joinflatshare.ui.register.photo.RegisterPhotoActivity;
import com.joinflatshare.utils.helper.CommonMethod;
import com.joinflatshare.utils.helper.CommonMethods;
import com.joinflatshare.utils.mixpanel.MixpanelUtils;
import com.joinflatshare.utils.system.DeviceInformationCollector;

import java.util.HashMap;

public class ProfileCreateActivity extends RegisterBaseActivity implements OnUiEventClick {
    protected ActivityProfileCreateBinding viewBind;
    public ProfileCreateDataBinder dataBinder;
    UserResponse response;
    User modelUser;
    protected boolean isNewUser = false;
    private ApiManager apiManager;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewBind = ActivityProfileCreateBinding.inflate(getLayoutInflater());
        setContentView(viewBind.getRoot());
        init();
    }

    private void init() {
        User loggedInUser = FlatShareApplication.Companion.getDbInstance().userDao().getUser();
        isNewUser = loggedInUser == null || loggedInUser.getName() == null ||
                loggedInUser.getName().getFirstName().isEmpty();

        if (isNewUser) {
            response = CommonMethods.getSerializable(getIntent(), "user", UserResponse.class);
            modelUser = response.getData();
        } else {
            modelUser = loggedInUser;
        }
        apiManager = new ApiManager(this);
        dataBinder = new ProfileCreateDataBinder(this, viewBind);
        new ProfileCreateListener(this, viewBind);
        dataBinder.setData();
    }

    void checkLocation() {
        apiManager.showProgress();
        getLocation(this,new GpsHandlerCallback() {
            @Override
            public void onLocationUpdate(double latitude, double longitude) {
                Loc loc = new Loc();
                loc.getCoordinates().add(longitude);
                loc.getCoordinates().add(latitude);
                modelUser.setLocation(new ModelLocation("My Location", loc));
                apiManager.hideProgress();
                updateUser();
            }

            @Override
            public void onLocationFailed() {
                apiManager.hideProgress();
                CommonMethods.makeToast("Could not proceed without user location");
            }
        });
    }

    void updateUser() {
        ApiManager apiManager = new ApiManager(this);
        apiManager.updateProfile(true, modelUser, response -> {
            new DeviceInformationCollector();
            MixpanelUtils.INSTANCE.sendToMixPanel("Member Joined");
            Intent intent = new Intent(ProfileCreateActivity.this, RegisterPhotoActivity.class);
            CommonMethod.INSTANCE.switchActivity(ProfileCreateActivity.this, intent, true);

            if (AppConstants.isSendbirdLive) {
                new ApplicationChatHandler().initialise(text -> {
                    if (text.equals("1")) {
                        SendBirdUser sendBirdUser = new SendBirdUser(ProfileCreateActivity.this);
                        HashMap<String, String> params = new HashMap<>();
                        params.put("user_id", modelUser.getId());
                        params.put("nickname", modelUser.getName().getFirstName() + " " + modelUser.getName().getLastName());
                        sendBirdUser.createUser(params, response1 -> {
                        });
                    }
                });
            }
            finishAffinity();
        });
    }

    @Override
    public void onClick(Intent intent, int requestCode) {
        if (requestCode == 1) {
            String type = intent.getStringExtra("type");
            if (type.equals(InterestsView.VIEW_TYPE_INTERESTS)) {
                viewBind.txtProfileInterest.setText(intent.getStringExtra("interest"));
                viewBind.txtInterestCount.setText("(" + intent.getIntExtra("count", 0) + "/5)");
            } else if (type.equals(InterestsView.VIEW_TYPE_LANGUAGES)) {
                viewBind.txtProfileLanguages.setText(intent.getStringExtra("interest"));
                viewBind.txtLanguageCount.setText("(" + intent.getIntExtra("count", 0) + "/3)");
            }
        }
    }
}

File: Flatshare Code/FlatShare 2.0/FlatshareCentral/src/main/java/com/joinflatshare/ui/register/create/ProfileCreateActivity.kt
package com.joinflatshare.ui.register.create

import android.content.Intent
import android.os.Bundle
import android.text.InputFilter
import android.text.Spanned
import com.joinflatshare.FlatShareApplication
import com.joinflatshare.FlatshareCentral.databinding.ActivityProfileCreateBinding
import com.joinflatshare.chat.ApplicationChatHandler
import com.joinflatshare.chat.SendBirdUser
import com.joinflatshare.chat.pojo.user.ModelChatUserResponse
import com.joinflatshare.constants.AppConstants
import com.joinflatshare.fcm.NotificationPermissionHandler
import com.joinflatshare.interfaces.OnUserFetched
import com.joinflatshare.pojo.user.User
import com.joinflatshare.pojo.user.UserResponse
import com.joinflatshare.ui.register.RegisterBaseActivity
import com.joinflatshare.ui.register.photo.RegisterPhotoActivity
import com.joinflatshare.ui.settings.SettingsActivity
import com.joinflatshare.utils.helper.CommonMethod
import com.joinflatshare.utils.helper.CommonMethods
import com.joinflatshare.utils.mixpanel.MixpanelUtils
import com.joinflatshare.utils.system.DeviceInformationCollector
import com.joinflatshare.webservice.api.WebserviceManager

/**
 * Created by debopam on 03/02/24
 */
class ProfileCreateActivity : RegisterBaseActivity() {
    private lateinit var viewBind: ActivityProfileCreateBinding
    var user: User? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBind = ActivityProfileCreateBinding.inflate(layoutInflater)
        setContentView(viewBind.root)
        MixpanelUtils.onScreenOpened("Onboarding Registration")
        user = FlatShareApplication.getDbInstance().userDao().getUser()
        if (user == null) {
            CommonMethod.logout(this)
            return
        }
        setFilter()
        ProfileCreateListener(this, viewBind)
    }

    private fun setFilter() {
        val filterText = arrayOfNulls<InputFilter>(1)
        filterText[0] =
            InputFilter { source: CharSequence, start: Int, end: Int, dest: Spanned?, dstart: Int, dend: Int ->
                if (end > start) {
                    val acceptedChars = charArrayOf(
                        'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm',
                        'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
                        'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
                        'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', ' '
                    )
                    for (index in start until end) {
                        if (!String(acceptedChars).contains(source[index].toString())) {
                            return@InputFilter if (source.length == 0) "" else source.subSequence(
                                0,
                                source.length - 1
                            )
                        }
                    }
                }
                null
            }
        viewBind.edtFname.filters = arrayOf(filterText[0])
        viewBind.edtLname.filters = arrayOf(filterText[0])
    }

    fun updateUser(modelUser: User?) {
        updateUser(modelUser, object : OnUserFetched {
            override fun userFetched(resp: UserResponse?) {
                CommonMethods.registerUser(resp)
                callAfterRegister(modelUser!!)
            }
        })
    }

    private fun callAfterRegister(modelUser: User) {
        // Collect Device Information
        DeviceInformationCollector()

        CommonMethod.sendUserToDB(modelUser)

        // Requesting Notification permission for Android 13
        NotificationPermissionHandler(this).showNotificationPermission {
            MixpanelUtils.sendToMixPanel("Registration Complete")
            val intent = Intent(this, RegisterPhotoActivity::class.java)
            CommonMethod.switchActivity(this, intent, false)
            finishAffinity()
        }

        //Register in Sendbird
        /*WebserviceManager.uiWebserviceHandler.showProgress(this)
        ApplicationChatHandler().initialise { text: String ->
            WebserviceManager.uiWebserviceHandler.hideProgress(this)
            if (text == "1") {
                registerSendbird(modelUser)
                // Requesting Notification permission for Android 13
                NotificationPermissionHandler(this).showNotificationPermission {
                    MixpanelUtils.sendToMixPanel("Registration Complete")
                    val intent = Intent(this, RegisterPhotoActivity::class.java)
                    CommonMethod.switchActivity(this, intent, false)
                    finishAffinity()
                }
            }
        }*/
    }

    private fun registerSendbird(modelUser: User) {
        val userName = modelUser.name!!.firstName + " " + modelUser.name!!.lastName
        CommonMethod.makeLog("SendBird User Register", userName + "  " + modelUser.id)
        val sendBirdUser = SendBirdUser(this@ProfileCreateActivity)
        val params =
            HashMap<String, String>()
        params["user_id"] = modelUser.id
        params["nickname"] = userName
        params["profile_url"] = ""
        sendBirdUser.createUser(
            params
        ) {
            // Update the User again since nickname is not registering
            params.remove("user_id")
            params.remove("profile_url")
            params["nickname"] = userName
            sendBirdUser.updateUser(
                user?.id,
                params
            ) { response: ModelChatUserResponse? -> }
        }
    }
}